assert/assertions.go

```diff
@@ -1931,16 +1934,16 @@ func EventuallyWithT(t TestingT, condition func(collect *CollectT), waitFor time
 			go func() {
 				collect := new(CollectT)
 				defer func() {
-					ch <- collect.errors
```

Comment:
Previously, we would hit this send on the channel because we hit this `defer` after the panic.

Now, with `runtime.Goexit`, this `defer` will not get called at all. Does that cause problems?

Reply:
https://pkg.go.dev/runtime#Goexit

> Goexit terminates the goroutine that calls it. No other goroutine is affected. Goexit runs all deferred calls before terminating the goroutine.

Unless the `FailNow` test won't pass

Reply:
Ah, I didn't know that bit about it running the deferred calls :+1: